src/LibraryViewExtension/LibraryViewCustomization.cs

```diff
@@ -127,5 +133,59 @@ public Stream ToJSONStream()
         {
             return root.ToJSONStream();
         }
+
+        /// <summary>
+        /// Notifies when a resource stream is registered
+        /// </summary>
+        public event Action<string, Stream> ResourceStreamRegistered;
+
+        /// <summary>
+        /// Registers a given resource stream for a given url. If registered 
+        /// successfully the requested resource is returned from the given stream.
+        /// </summary>
+        /// <param name="urlpath">relative path of url including 
+        /// extension of the resource, e.g. /myicons/xicon.png</param>
+        /// <param name="resource">resource data stream</param>
+        /// <returns>True if the operation was successful</returns>
```

Comment:
Since this class has been made public, requirements for its documentation become more stringent. Perhaps we should have the purpose of this method listed in `summary` section, then also describe what kinds of values will fail the call for each parameter (e.g. it must have extension). It will be worthwhile to mention `ResourceStreamRegistered` being called only for successful registration.